pkg/loader/compose/v1v2.go

```diff
@@ -277,3 +322,80 @@ func checkLabelsPorts(noOfPort int, labels string, svcName string) error {
 	}
 	return nil
 }
+
+func findDuplicateVol(vols kobject.Volumes, volumes []kobject.Volumes) bool {
+	for _, v := range volumes {
+		if vols.MountPath == v.MountPath {
+			return true
+		}
+	}
+	return false
+}
+
+func handleVolume(svcName string, komposeObject kobject.KomposeObject) (volume []kobject.Volumes) {
+
+	if komposeObject.ServiceConfigs[svcName].VolumesFrom != nil {
+		for _, depVol := range komposeObject.ServiceConfigs[svcName].VolumesFrom {
+
+			dVols := handleVolume(depVol, komposeObject)
+			var cVols []kobject.Volumes
+			for _, v := range komposeObject.ServiceConfigs[svcName].Volumes {
+				vol := kobject.Volumes{
+					SvcName:   svcName,
+					MountPath: v,
+				}
+				cVols = append(cVols, vol)
+			}
+			for _, cv := range cVols {
+				i, dv := getVol(cv, dVols)
+				if i == true {
+					if dv.VFrom == "" {
+						cv.VFrom = dv.SvcName
+						cv.SvcName = dv.SvcName
+					} else {
+						cv.VFrom = dv.VFrom
+						cv.SvcName = dv.SvcName
+					}
+				}
+				volume = append(volume, cv)
+			}
+
+			for _, dv := range dVols {
+				if checkvolpresent(dv, cVols) {
+					dv.VFrom = dv.SvcName
+					volume = append(volume, dv)
+				}
+
+			}
+
+		}
+	} else {
+		for _, v := range komposeObject.ServiceConfigs[svcName].Volumes {
+			vol := kobject.Volumes{
+				SvcName:   svcName,
+				MountPath: v,
+			}
+			volume = append(volume, vol)
+
+		}
+	}
+	return volume
+}
+
+func checkvolpresent(tofind kobject.Volumes, cVols []kobject.Volumes) bool {
+	for _, cv := range cVols {
+		if tofind.MountPath != cv.MountPath {
+			return true
+		}
+	}
+	return false
+}
+
+func getVol(tofind kobject.Volumes, dVols []kobject.Volumes) (bool, kobject.Volumes) {
```

Comment:
This is a generic function that searches array of volumes without any bias on what the array of volumes is so maybe name it better, like `s/dVols/vols/`